bagitobjecttransfer/recordtransfer/storage.py

```diff
@@ -1,25 +1,53 @@
+from pathlib import Path
+
 from django.conf import settings
-from django.core.files.storage import get_storage_class
+from django.core.files.storage import FileSystemStorage
 
-Storage = get_storage_class()
 
-class OverwriteStorage(Storage):
-    ''' Overwrites files in storage if they have the same name.
+class OverwriteStorage(FileSystemStorage):
+    """Overwrites files in storage if they have the same name.
 
     Django's default method of storing files named the same thing is to append a unique suffix to
     the file name and saving the duplicate file there. This class overrides that behviour and
     overwrites any file being saved with a path that already exists.
-    '''
+    """
+
     def _save(self, name, content):
         self.delete(name)
         return super()._save(name, content)
 
     def get_available_name(self, name, max_length=None):
         return name
 
-class UploadedFileStorage(Storage):
-    ''' Stores files in UPLOAD_STORAGE_FOLDER
-    '''
-    def __init__(self, **kwargs):
-        kwargs['location'] = settings.UPLOAD_STORAGE_FOLDER
+
+class MediaFileStorage(FileSystemStorage):
+    """Base class for custom file storage."""
+
+    def __init__(self, location: str, **kwargs):
+        """Set the location of the storage. The location has to be a subdirectory of MEDIA_ROOT."""
+        # Check if the location is a subdirectory of MEDIA_ROOT
+        media_root = Path(settings.MEDIA_ROOT).resolve()
+        location_path = Path(location).resolve()
+
+        if not location_path.is_relative_to(media_root):
+            raise ValueError("The location must be a subdirectory of MEDIA_ROOT")
+        kwargs["location"] = str(location)
         super().__init__(**kwargs)
+
+    def url(self, name: str) -> str:
+        """Generate the URL based on MEDIA_URL and the relative path."""
+        return f"/{Path(self.base_location).relative_to(settings.BASE_DIR) / name}"
```

Comment:
This is technically a URL, so we should use `urljoin` here. As a benefit, this is platform-agnostic and won't create a path with backslasehs like the current code is doing in Windows.

```suggestion
        return urljoin(self.base_url, name)
```

Reply:
Using `as_posix()` on `base_url` `Path` instead of using `url_join`

Addressed by: https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/292/commits/152ecf548e49cc79804dcac70cb6b3d58491026b